### ISSUE TITLE
Glowpoints on subsystems broken in ship lab

### ISSUE THREAD

**Issue by chief1983** [scp-fs2open/fs2open.github.com#1206]
In the video at https://youtu.be/QbhOd6WM31g you can see that the glowpoints do now follow the Faustus' rotating subsystems in the ship lab.  I don't believe this bug exists in game, and has existed in the ship lab since 3.7.2 or earlier.  I can't currently test before that without downloading an older set of MediaVP files.

**Comment by asarium** [comment]
I looked at the glowpoint code and that code is very hard to follow. There is a lot of code that handles ship specific stuff which may contain handling of subsystem rotation and offset. Since we don't actually use the ship instance in the lab this might explain why the glowpoints don't rotate.

**Comment by TRBlount** [comment]
I believe the issue lies in the fact that the glowpoint rendering functions are passed null ship pointers so they can't access the submodel_rotation portions of the code.

This can be fixed quite easily, we just need to set the objnum of the model_render_params used in labviewer_model_render(), which is -1 by default. The question is how do we decide which objnum to pass to the render parameters.

**Comment by asarium** [comment]
That doesn't work because I just fixed another issue with that by making sure it's -1 everywhere. The lab uses a ship object for some things but for most other things it manipulates the model directly. The proper solution would be to convert everything to the new object based system but that is quite a bit of work since the existing code assumes that the custom subsystem data is used.

@Goober5000 You did some changes to that code (at least the comments say that). Do you have any idea how to fix this properly?

**Comment by chief1983** [comment]
@The-E @asarium any chance you guys had insight for this one while fixing #1264?

**Comment by chief1983** [comment]
Also @Goober5000 ping?

**Comment by The-E** [comment]
I can take a look, but I believe these issues to be rather different. After all, #1264 was about functionality that was never implemented, this is stuff that I distinctly recall working at some point.

**Comment by chief1983** [comment]
Ah, I would have thought it the other way around, the glowpoints seem to have broken for quite a while back.  Not sure I can find an instance of them working in fact.

**Comment by Goober5000** [comment]
Here's a late response to that ping; I'm slowly going through my backlog of emails.

@asarium Where is the code that has comments that say I made changes?

**Comment by asarium** [comment]
I can't find the comments anymore and I don't remember which part I was referring to specifically.

**Comment by Goober5000** [comment]
Well, I can tell you that when I was working on the look_at fixes, it was necessary to get @SamuelCho's help in completing the conversion of the submodel rotation code from the old method to the new method.  It's possible that he made some simplifications to get the lab to work without actually handling the rotation.

**Comment by Goober5000** [comment]
I just checked and this is now fixed.  Not sure when it happened.
